From 9d905bbf5b1e021c1b9b5c3ce3399a91deb23eb7 Mon Sep 17 00:00:00 2001
From: Hien Dang <hien.dang.eb@renesas.com>
Date: Tue, 23 Aug 2016 10:33:19 +0700
Subject: [PATCH 177/909] serial: sh-sci: Use pm_runtime_get/put when resume

commit db75c1ad6481e00b9b305e820cc5109a4b329634 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Due to the commit '39dd0f234fc37d ("PM / Domains: Allow runtime PM during
system PM phases")', runtime PM status might be suspended during system
suspend phase. Therefore, at system suspend phase, pm_runtime_get()/put()
are necessary invoked when accessing the hardware.

This modification,in system resume phase, is for symmetric with above
situation. Moreover, it enhances stability of system in case that
upstream kernel may update to enable device can change runtime suspend
state during system resume phase.

Signed-off-by: Hien Dang <hien.dang.eb@renesas.com>
Signed-off-by: Hiromitsu Yamasaki <hiromitsu.yamasaki.ym@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/sh-sci.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/serial/sh-sci.c b/drivers/tty/serial/sh-sci.c
index e7f9cca632a4..d9a5d8f2aaa6 100644
--- a/drivers/tty/serial/sh-sci.c
+++ b/drivers/tty/serial/sh-sci.c
@@ -3278,8 +3278,11 @@ static __maybe_unused int sci_resume(struct device *dev)
 {
 	struct sci_port *sport = dev_get_drvdata(dev);
 
-	if (sport)
+	if (sport) {
+		pm_runtime_get_sync(sport->port.dev);
 		uart_resume_port(&sci_uart_driver, &sport->port);
+		pm_runtime_put(sport->port.dev);
+	}
 
 	return 0;
 }
-- 
2.17.1

